src/algebra/direct_sum/ring.lean

```diff
@@ -122,6 +122,16 @@ class gcomm_ring [add_comm_monoid ι] [Π i, add_comm_group (A i)] extends
 
 end defs
 
+/-- A graded version of `semiring.to_module`. -/
+instance gsemiring.to_gmodule (A : ι → Type*)
+  [add_monoid ι] [Π (i : ι), add_comm_monoid (A i)] [gsemiring A] :
+  graded_monoid.gmodule A A :=
+{ smul_add := λ _ _, gsemiring.mul_add,
+  smul_zero := λ i j, gsemiring.mul_zero,
+  add_smul := λ i j, gsemiring.add_mul,
+  zero_smul := λ i j, gsemiring.zero_mul,
+  ..graded_monoid.gmonoid.to_gmul_action A }
```

Comment:
I fixed this error [here](https://github.com/leanprover-community/mathlib/commit/ba994c5b8d4b3b95197bf5dfbba3a391ecfaa36a) by moving this declaration to the graded_module file. If everyone is happy we can merge my branch then send this PR to bors.

Reply:
Also see the [latest commit](https://github.com/leanprover-community/mathlib/commit/71d4efd74657593064f791206c81f05eb0bf746f) that addressed @kbuzzard's comments.